src/lua/compat.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2022, Tarantool AUTHORS, please see AUTHORS file.
+ */
+
+#include <lua/compat.h>
+
+static const char double_newline[] = "\n\n";
+static const char single_newline[] = "\n";
+
+char *FORCE_LITERAL_SUBSTRING = NULL;
+
+int
+yaml_pretty_multiline_toggle(struct lua_State *L)
+{
+	int top = lua_gettop(L);
+	assert(lua_type(L, top) == LUA_TBOOLEAN);
```

Comment:
The error is caused by `isboolean()`/`gettop()` because there is no argument. When called with a wrong arguments, the assert works correctly. As it is an internal error, assert should be fine.
```
Tarantool 2.10.0-beta1-1204-gf37572732
type 'help' for interactive help
tarantool> cmp = require("compat_internals")
---
...

tarantool> cmp.yaml_pretty_multiline_toggle('string')
tarantool: /home/ubuntu/Code/tarantool-yaml-prettier-multiline-output/src/lua/compat.c:17: yaml_pretty_multiline_toggle: Assertion `lua_isboolean(L, -1) && "boolean argument expected"' failed.
[1]    1247382 IOT instruction (core dumped)  ./src/tarantool
```